{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test Protocol for the Benchmark Analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The revised idea for the benchmark analysis pipeline, is to ask from the user the path to the directory that holds data files constituting the benchmark data. This input is fed into the pipeline and the results (tool outputs, visualizations are engendered and stored on physical drive as well). "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let us commence by instituting the work environment. The tools in question are **Chipenrich**, **Broadenrich**, **Seq2pathway**, **Enrichr**, and **GREAT**. We now install the tools and load their respective libraries. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following function code installs the R packages for Chipenrich, Broadenrich, and Seq2pathway. The tools *Enrichr* and *GREAT* are only available as web-interfaces and so the input-output for these have to be catered externally. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installing Tool Packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.0 (2019-04-26)\n",
      "\n",
      "Installing package(s) 'seq2pathway'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The downloaded binary packages are in\n",
      "\t/var/folders/hm/c3_fjypn62v5xh5b5ygv267m0000gn/T//RtmpE6p2Np/downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'Rdpack', 'bibtex', 'broom', 'BH', 'DBI', 'DT', 'MASS',\n",
      "  'RSQLite', 'SparseM', 'boot', 'caTools', 'digest', 'e1071', 'foreign',\n",
      "  'latticeExtra', 'mime', 'pillar', 'quantreg', 'recipes', 'repr',\n",
      "  'reticulate', 'robust', 'rsconnect', 'vctrs'\n",
      "\n",
      "Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.0 (2019-04-26)\n",
      "\n",
      "Installing package(s) 'broadenrich'\n",
      "\n",
      "Warning message:\n",
      "“package ‘broadenrich’ is not available (for R version 3.6.0)”\n",
      "Old packages: 'Rdpack', 'bibtex', 'broom', 'BH', 'DBI', 'DT', 'MASS',\n",
      "  'RSQLite', 'SparseM', 'boot', 'caTools', 'digest', 'e1071', 'foreign',\n",
      "  'latticeExtra', 'mime', 'pillar', 'quantreg', 'recipes', 'repr',\n",
      "  'reticulate', 'robust', 'rsconnect', 'vctrs'\n",
      "\n",
      "Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.0 (2019-04-26)\n",
      "\n",
      "Installing package(s) 'chipenrich'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The downloaded binary packages are in\n",
      "\t/var/folders/hm/c3_fjypn62v5xh5b5ygv267m0000gn/T//RtmpE6p2Np/downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'Rdpack', 'bibtex', 'broom', 'BH', 'DBI', 'DT', 'MASS',\n",
      "  'RSQLite', 'SparseM', 'boot', 'caTools', 'digest', 'e1071', 'foreign',\n",
      "  'latticeExtra', 'mime', 'pillar', 'quantreg', 'recipes', 'repr',\n",
      "  'reticulate', 'robust', 'rsconnect', 'vctrs'\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Warning message:\n",
      "“'memory.size()' is Windows-specific”\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "Inf"
      ],
      "text/latex": [
       "Inf"
      ],
      "text/markdown": [
       "Inf"
      ],
      "text/plain": [
       "[1] Inf"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "source(\"./protocolFunctions/installPackagesBenchmark.R\")\n",
    "installPackagesBenchmark()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that we have all the R-based tools synced in, we shall now move towards assembling the test data. The BED files for the respective samples are available at a local folder. However, for the fuller version of the analysis pipeline I shall have the user input the path for the folder that holds the data files for the benchmark data.  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Installing Support Packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Devtools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.0 (2019-04-26)\n",
      "\n",
      "Installing package(s) 'devtools'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The downloaded binary packages are in\n",
      "\t/var/folders/hm/c3_fjypn62v5xh5b5ygv267m0000gn/T//RtmpE6p2Np/downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'Rdpack', 'bibtex', 'broom', 'BH', 'DBI', 'DT', 'MASS',\n",
      "  'RSQLite', 'SparseM', 'boot', 'caTools', 'digest', 'e1071', 'foreign',\n",
      "  'latticeExtra', 'mime', 'pillar', 'quantreg', 'recipes', 'repr',\n",
      "  'reticulate', 'robust', 'rsconnect', 'vctrs'\n",
      "\n",
      "Loading required package: usethis\n",
      "\n"
     ]
    }
   ],
   "source": [
    "## 'devtools' provides multiple utilitarian functionalities. Let us install this package as well.\n",
    "\n",
    "BiocManager::install('devtools')\n",
    "library(devtools)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GenomicRanges"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Bioconductor version 3.9 (BiocManager 1.30.10), R 3.6.0 (2019-04-26)\n",
      "\n",
      "Installing package(s) 'GenomicRanges'\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The downloaded binary packages are in\n",
      "\t/var/folders/hm/c3_fjypn62v5xh5b5ygv267m0000gn/T//RtmpE6p2Np/downloaded_packages\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Old packages: 'Rdpack', 'bibtex', 'broom', 'BH', 'DBI', 'DT', 'MASS',\n",
      "  'RSQLite', 'SparseM', 'boot', 'caTools', 'digest', 'e1071', 'foreign',\n",
      "  'latticeExtra', 'mime', 'pillar', 'quantreg', 'recipes', 'repr',\n",
      "  'reticulate', 'robust', 'rsconnect', 'vctrs'\n",
      "\n",
      "Warning message:\n",
      "“package ‘GenomicRanges’ was built under R version 3.6.1”\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Warning message:\n",
      "“package ‘S4Vectors’ was built under R version 3.6.1”\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Warning message:\n",
      "“package ‘IRanges’ was built under R version 3.6.1”\n",
      "Loading required package: GenomeInfoDb\n",
      "\n"
     ]
    }
   ],
   "source": [
    "BiocManager::install('GenomicRanges')\n",
    "library(GenomicRanges)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Importing Data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The function **dataImportClean** solicits the directory path from the user that holds the constituent BED files for the benchmark dataset. The input (as depicted below) must be made as a character expression followed by a backslash. The files must have the fundamental attributes of a genomic region, viz. *chrom*, *start*, and *end*. The data is sourced as GRanges objects for subsequent manipulations. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The dataset in question has the following profile, amongst other attributes. It is a mix of assorted diseases that act as our target pathways for a given instance of comparison of enrichment tools. (See https://doi.org/10.1371/journal.pone.0079217 for methodological reference) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in parse(text = x, srcfile = src): <text>:1:2: unexpected '['\n1: ![\n     ^\n",
     "output_type": "error",
     "traceback": [
      "Error in parse(text = x, srcfile = src): <text>:1:2: unexpected '['\n1: ![\n     ^\nTraceback:\n"
     ]
    }
   ],
   "source": [
    "![](testData.png)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32m✔\u001b[39m Setting active project to \u001b[34m'/Users/mei/Desktop/GSABenchmarkTestAnalysis'\u001b[39m\n"
     ]
    }
   ],
   "source": [
    "source(\"./protocolFunctions/dataImportClean.R\")\n",
    "dataImportClean(\"./testData/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The list variable 'samplesInBED' holds the data files. Let us look at the data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "samplesInBED <- readRDS(\"samplesInBED.rds\")\n",
    "ChIPSeqSamples <- readRDS(\"ChIPSeqSamples.rds\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRangesList object of length 10:\n",
       "$GSE84618 \n",
       "GRanges object with 9789 ranges and 0 metadata columns:\n",
       "         seqnames              ranges strand\n",
       "            <Rle>           <IRanges>  <Rle>\n",
       "     [1]     chr1 181153247-181153976      *\n",
       "     [2]     chr1   54590230-54590888      *\n",
       "     [3]     chr1 183400489-183400671      *\n",
       "     [4]     chr1   55420208-55420369      *\n",
       "     [5]     chr1   20114346-20114575      *\n",
       "     ...      ...                 ...    ...\n",
       "  [9785]     chrX   68346920-68347399      *\n",
       "  [9786]     chrX   70474572-70474901      *\n",
       "  [9787]     chrX   53105208-53105440      *\n",
       "  [9788]     chrX   20065462-20065651      *\n",
       "  [9789]     chrX   77395786-77395955      *\n",
       "\n",
       "...\n",
       "<9 more elements>\n",
       "-------\n",
       "seqinfo: 64 sequences from hg38 genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "samplesInBED"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So, we see that we have 10 samples listed in GRanges format. These will be our input to the tools and the basis for comparison."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Executing Chipenrich, Broadenrich, and Seq2pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "While executing the following function we make an attempt to save the results as R objects and concurrently remove them from active memory for sufficiency. This wrapper script holds the executives for the three tools with specific parameters."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSE84618.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 09:38:53.071541\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 09:39:13.737955\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 09:39:16.400911\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM1847178.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 09:44:35.239472\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 09:44:55.212420\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 09:45:00.321482\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2058021.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 09:51:59.340023\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 09:52:18.782591\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 09:52:46.553804\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2058022.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 10:00:33.794035\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 10:00:53.341784\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 10:01:10.678266\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2058023.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 10:09:00.015946\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 10:09:21.080501\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 10:09:48.216724\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2101436.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 10:18:16.079567\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 10:18:36.549075\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Un_gl000224 Chromosome not registered\"           \n",
      "[7] \"Finish Annotation\"                               \n",
      "[8] \"python process end: 2020-01-06 10:18:36.646784\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2101437.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"python process start: 2020-01-06 10:20:51.431282\"\n",
      " [2] \"Load Reference\"                                  \n",
      " [3] \"Check Reference files\"                           \n",
      " [4] \"fixed reference done: 2020-01-06 10:21:11.415655\"\n",
      " [5] \"Start Annotation\"                                \n",
      " [6] \"1_gl000191_random Chromosome not registered\"     \n",
      " [7] \"1_gl000191_random Chromosome not registered\"     \n",
      " [8] \"1_gl000192_random Chromosome not registered\"     \n",
      " [9] \"1_gl000192_random Chromosome not registered\"     \n",
      "[10] \"1_gl000192_random Chromosome not registered\"     \n",
      "[11] \"1_gl000192_random Chromosome not registered\"     \n",
      "[12] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[13] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[14] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[15] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[16] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[17] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[18] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[19] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[20] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[21] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[22] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[23] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[24] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[25] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[26] \"17_gl000204_random Chromosome not registered\"    \n",
      "[27] \"17_gl000205_random Chromosome not registered\"    \n",
      "[28] \"17_gl000205_random Chromosome not registered\"    \n",
      "[29] \"19_gl000208_random Chromosome not registered\"    \n",
      "[30] \"4_gl000193_random Chromosome not registered\"     \n",
      "[31] \"4_gl000194_random Chromosome not registered\"     \n",
      "[32] \"6_cox_hap2 Chromosome not registered\"            \n",
      "[33] \"6_mann_hap4 Chromosome not registered\"           \n",
      "[34] \"6_mcf_hap5 Chromosome not registered\"            \n",
      "[35] \"6_mcf_hap5 Chromosome not registered\"            \n",
      "[36] \"6_mcf_hap5 Chromosome not registered\"            \n",
      "[37] \"6_qbl_hap6 Chromosome not registered\"            \n",
      "[38] \"7_gl000195_random Chromosome not registered\"     \n",
      "[39] \"7_gl000195_random Chromosome not registered\"     \n",
      "[40] \"7_gl000195_random Chromosome not registered\"     \n",
      "[41] \"7_gl000195_random Chromosome not registered\"     \n",
      "[42] \"Un_gl000211 Chromosome not registered\"           \n",
      "[43] \"Un_gl000214 Chromosome not registered\"           \n",
      "[44] \"Un_gl000214 Chromosome not registered\"           \n",
      "[45] \"Un_gl000214 Chromosome not registered\"           \n",
      "[46] \"Un_gl000214 Chromosome not registered\"           \n",
      "[47] \"Un_gl000217 Chromosome not registered\"           \n",
      "[48] \"Un_gl000218 Chromosome not registered\"           \n",
      "[49] \"Un_gl000218 Chromosome not registered\"           \n",
      "[50] \"Un_gl000218 Chromosome not registered\"           \n",
      "[51] \"Un_gl000219 Chromosome not registered\"           \n",
      "[52] \"Un_gl000219 Chromosome not registered\"           \n",
      "[53] \"Un_gl000219 Chromosome not registered\"           \n",
      "[54] \"Un_gl000220 Chromosome not registered\"           \n",
      "[55] \"Un_gl000225 Chromosome not registered\"           \n",
      "[56] \"Un_gl000237 Chromosome not registered\"           \n",
      "[57] \"Un_gl000240 Chromosome not registered\"           \n",
      "[58] \"Un_gl000240 Chromosome not registered\"           \n",
      "[59] \"Un_gl000240 Chromosome not registered\"           \n",
      "[60] \"Un_gl000241 Chromosome not registered\"           \n",
      "[61] \"Finish Annotation\"                               \n",
      "[62] \"python process end: 2020-01-06 10:21:20.825395\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2101438.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  [1] \"python process start: 2020-01-06 10:29:31.210499\"\n",
      "  [2] \"Load Reference\"                                  \n",
      "  [3] \"Check Reference files\"                           \n",
      "  [4] \"fixed reference done: 2020-01-06 10:29:51.426844\"\n",
      "  [5] \"Start Annotation\"                                \n",
      "  [6] \"17_gl000203_random Chromosome not registered\"    \n",
      "  [7] \"17_gl000205_random Chromosome not registered\"    \n",
      "  [8] \"17_gl000205_random Chromosome not registered\"    \n",
      "  [9] \"17_gl000205_random Chromosome not registered\"    \n",
      " [10] \"17_gl000205_random Chromosome not registered\"    \n",
      " [11] \"17_gl000205_random Chromosome not registered\"    \n",
      " [12] \"17_gl000205_random Chromosome not registered\"    \n",
      " [13] \"17_gl000205_random Chromosome not registered\"    \n",
      " [14] \"17_gl000205_random Chromosome not registered\"    \n",
      " [15] \"17_gl000205_random Chromosome not registered\"    \n",
      " [16] \"17_gl000205_random Chromosome not registered\"    \n",
      " [17] \"17_gl000205_random Chromosome not registered\"    \n",
      " [18] \"17_gl000205_random Chromosome not registered\"    \n",
      " [19] \"17_gl000205_random Chromosome not registered\"    \n",
      " [20] \"17_gl000205_random Chromosome not registered\"    \n",
      " [21] \"17_gl000205_random Chromosome not registered\"    \n",
      " [22] \"19_gl000208_random Chromosome not registered\"    \n",
      " [23] \"19_gl000208_random Chromosome not registered\"    \n",
      " [24] \"19_gl000208_random Chromosome not registered\"    \n",
      " [25] \"19_gl000208_random Chromosome not registered\"    \n",
      " [26] \"19_gl000208_random Chromosome not registered\"    \n",
      " [27] \"19_gl000208_random Chromosome not registered\"    \n",
      " [28] \"19_gl000208_random Chromosome not registered\"    \n",
      " [29] \"19_gl000208_random Chromosome not registered\"    \n",
      " [30] \"19_gl000208_random Chromosome not registered\"    \n",
      " [31] \"19_gl000208_random Chromosome not registered\"    \n",
      " [32] \"19_gl000208_random Chromosome not registered\"    \n",
      " [33] \"19_gl000208_random Chromosome not registered\"    \n",
      " [34] \"19_gl000208_random Chromosome not registered\"    \n",
      " [35] \"19_gl000208_random Chromosome not registered\"    \n",
      " [36] \"19_gl000208_random Chromosome not registered\"    \n",
      " [37] \"19_gl000208_random Chromosome not registered\"    \n",
      " [38] \"19_gl000208_random Chromosome not registered\"    \n",
      " [39] \"19_gl000208_random Chromosome not registered\"    \n",
      " [40] \"19_gl000208_random Chromosome not registered\"    \n",
      " [41] \"19_gl000208_random Chromosome not registered\"    \n",
      " [42] \"19_gl000208_random Chromosome not registered\"    \n",
      " [43] \"19_gl000208_random Chromosome not registered\"    \n",
      " [44] \"4_gl000193_random Chromosome not registered\"     \n",
      " [45] \"4_gl000193_random Chromosome not registered\"     \n",
      " [46] \"4_gl000193_random Chromosome not registered\"     \n",
      " [47] \"4_gl000193_random Chromosome not registered\"     \n",
      " [48] \"4_gl000193_random Chromosome not registered\"     \n",
      " [49] \"4_gl000193_random Chromosome not registered\"     \n",
      " [50] \"4_gl000193_random Chromosome not registered\"     \n",
      " [51] \"4_gl000193_random Chromosome not registered\"     \n",
      " [52] \"4_gl000193_random Chromosome not registered\"     \n",
      " [53] \"4_gl000193_random Chromosome not registered\"     \n",
      " [54] \"4_gl000193_random Chromosome not registered\"     \n",
      " [55] \"4_gl000193_random Chromosome not registered\"     \n",
      " [56] \"4_gl000193_random Chromosome not registered\"     \n",
      " [57] \"4_gl000193_random Chromosome not registered\"     \n",
      " [58] \"4_gl000193_random Chromosome not registered\"     \n",
      " [59] \"4_gl000193_random Chromosome not registered\"     \n",
      " [60] \"4_gl000193_random Chromosome not registered\"     \n",
      " [61] \"4_gl000193_random Chromosome not registered\"     \n",
      " [62] \"4_gl000193_random Chromosome not registered\"     \n",
      " [63] \"4_gl000193_random Chromosome not registered\"     \n",
      " [64] \"4_gl000193_random Chromosome not registered\"     \n",
      " [65] \"4_gl000193_random Chromosome not registered\"     \n",
      " [66] \"4_gl000193_random Chromosome not registered\"     \n",
      " [67] \"4_gl000193_random Chromosome not registered\"     \n",
      " [68] \"4_gl000193_random Chromosome not registered\"     \n",
      " [69] \"4_gl000193_random Chromosome not registered\"     \n",
      " [70] \"4_gl000193_random Chromosome not registered\"     \n",
      " [71] \"4_gl000193_random Chromosome not registered\"     \n",
      " [72] \"4_gl000193_random Chromosome not registered\"     \n",
      " [73] \"4_gl000193_random Chromosome not registered\"     \n",
      " [74] \"4_gl000193_random Chromosome not registered\"     \n",
      " [75] \"4_gl000193_random Chromosome not registered\"     \n",
      " [76] \"4_gl000193_random Chromosome not registered\"     \n",
      " [77] \"4_gl000193_random Chromosome not registered\"     \n",
      " [78] \"4_gl000193_random Chromosome not registered\"     \n",
      " [79] \"4_gl000193_random Chromosome not registered\"     \n",
      " [80] \"4_gl000193_random Chromosome not registered\"     \n",
      " [81] \"4_gl000193_random Chromosome not registered\"     \n",
      " [82] \"4_gl000193_random Chromosome not registered\"     \n",
      " [83] \"4_gl000193_random Chromosome not registered\"     \n",
      " [84] \"4_gl000193_random Chromosome not registered\"     \n",
      " [85] \"4_gl000193_random Chromosome not registered\"     \n",
      " [86] \"4_gl000193_random Chromosome not registered\"     \n",
      " [87] \"4_gl000193_random Chromosome not registered\"     \n",
      " [88] \"4_gl000193_random Chromosome not registered\"     \n",
      " [89] \"4_gl000193_random Chromosome not registered\"     \n",
      " [90] \"4_gl000193_random Chromosome not registered\"     \n",
      " [91] \"4_gl000193_random Chromosome not registered\"     \n",
      " [92] \"4_gl000193_random Chromosome not registered\"     \n",
      " [93] \"4_gl000193_random Chromosome not registered\"     \n",
      " [94] \"4_gl000193_random Chromosome not registered\"     \n",
      " [95] \"4_gl000193_random Chromosome not registered\"     \n",
      " [96] \"4_gl000193_random Chromosome not registered\"     \n",
      " [97] \"4_gl000193_random Chromosome not registered\"     \n",
      " [98] \"4_gl000193_random Chromosome not registered\"     \n",
      " [99] \"4_gl000193_random Chromosome not registered\"     \n",
      "[100] \"4_gl000193_random Chromosome not registered\"     \n",
      "[101] \"4_gl000193_random Chromosome not registered\"     \n",
      "[102] \"4_gl000193_random Chromosome not registered\"     \n",
      "[103] \"4_gl000193_random Chromosome not registered\"     \n",
      "[104] \"4_gl000194_random Chromosome not registered\"     \n",
      "[105] \"4_gl000194_random Chromosome not registered\"     \n",
      "[106] \"4_gl000194_random Chromosome not registered\"     \n",
      "[107] \"4_gl000194_random Chromosome not registered\"     \n",
      "[108] \"7_gl000195_random Chromosome not registered\"     \n",
      "[109] \"7_gl000195_random Chromosome not registered\"     \n",
      "[110] \"7_gl000195_random Chromosome not registered\"     \n",
      "[111] \"7_gl000195_random Chromosome not registered\"     \n",
      "[112] \"9_gl000198_random Chromosome not registered\"     \n",
      "[113] \"9_gl000198_random Chromosome not registered\"     \n",
      "[114] \"9_gl000199_random Chromosome not registered\"     \n",
      "[115] \"9_gl000199_random Chromosome not registered\"     \n",
      "[116] \"Un_gl000211 Chromosome not registered\"           \n",
      "[117] \"Un_gl000211 Chromosome not registered\"           \n",
      "[118] \"Un_gl000211 Chromosome not registered\"           \n",
      "[119] \"Un_gl000211 Chromosome not registered\"           \n",
      "[120] \"Un_gl000211 Chromosome not registered\"           \n",
      "[121] \"Un_gl000211 Chromosome not registered\"           \n",
      "[122] \"Un_gl000214 Chromosome not registered\"           \n",
      "[123] \"Un_gl000214 Chromosome not registered\"           \n",
      "[124] \"Un_gl000214 Chromosome not registered\"           \n",
      "[125] \"Un_gl000214 Chromosome not registered\"           \n",
      "[126] \"Un_gl000214 Chromosome not registered\"           \n",
      "[127] \"Un_gl000214 Chromosome not registered\"           \n",
      "[128] \"Un_gl000214 Chromosome not registered\"           \n",
      "[129] \"Un_gl000214 Chromosome not registered\"           \n",
      "[130] \"Un_gl000214 Chromosome not registered\"           \n",
      "[131] \"Un_gl000214 Chromosome not registered\"           \n",
      "[132] \"Un_gl000214 Chromosome not registered\"           \n",
      "[133] \"Un_gl000214 Chromosome not registered\"           \n",
      "[134] \"Un_gl000214 Chromosome not registered\"           \n",
      "[135] \"Un_gl000214 Chromosome not registered\"           \n",
      "[136] \"Un_gl000214 Chromosome not registered\"           \n",
      "[137] \"Un_gl000214 Chromosome not registered\"           \n",
      "[138] \"Un_gl000214 Chromosome not registered\"           \n",
      "[139] \"Un_gl000214 Chromosome not registered\"           \n",
      "[140] \"Un_gl000214 Chromosome not registered\"           \n",
      "[141] \"Un_gl000214 Chromosome not registered\"           \n",
      "[142] \"Un_gl000214 Chromosome not registered\"           \n",
      "[143] \"Un_gl000214 Chromosome not registered\"           \n",
      "[144] \"Un_gl000214 Chromosome not registered\"           \n",
      "[145] \"Un_gl000214 Chromosome not registered\"           \n",
      "[146] \"Un_gl000214 Chromosome not registered\"           \n",
      "[147] \"Un_gl000214 Chromosome not registered\"           \n",
      "[148] \"Un_gl000214 Chromosome not registered\"           \n",
      "[149] \"Un_gl000214 Chromosome not registered\"           \n",
      "[150] \"Un_gl000214 Chromosome not registered\"           \n",
      "[151] \"Un_gl000214 Chromosome not registered\"           \n",
      "[152] \"Un_gl000214 Chromosome not registered\"           \n",
      "[153] \"Un_gl000214 Chromosome not registered\"           \n",
      "[154] \"Un_gl000214 Chromosome not registered\"           \n",
      "[155] \"Un_gl000214 Chromosome not registered\"           \n",
      "[156] \"Un_gl000214 Chromosome not registered\"           \n",
      "[157] \"Un_gl000214 Chromosome not registered\"           \n",
      "[158] \"Un_gl000214 Chromosome not registered\"           \n",
      "[159] \"Un_gl000214 Chromosome not registered\"           \n",
      "[160] \"Un_gl000214 Chromosome not registered\"           \n",
      "[161] \"Un_gl000214 Chromosome not registered\"           \n",
      "[162] \"Un_gl000214 Chromosome not registered\"           \n",
      "[163] \"Un_gl000216 Chromosome not registered\"           \n",
      "[164] \"Un_gl000216 Chromosome not registered\"           \n",
      "[165] \"Un_gl000216 Chromosome not registered\"           \n",
      "[166] \"Un_gl000216 Chromosome not registered\"           \n",
      "[167] \"Un_gl000216 Chromosome not registered\"           \n",
      "[168] \"Un_gl000216 Chromosome not registered\"           \n",
      "[169] \"Un_gl000216 Chromosome not registered\"           \n",
      "[170] \"Un_gl000216 Chromosome not registered\"           \n",
      "[171] \"Un_gl000216 Chromosome not registered\"           \n",
      "[172] \"Un_gl000216 Chromosome not registered\"           \n",
      "[173] \"Un_gl000216 Chromosome not registered\"           \n",
      "[174] \"Un_gl000216 Chromosome not registered\"           \n",
      "[175] \"Un_gl000216 Chromosome not registered\"           \n",
      "[176] \"Un_gl000216 Chromosome not registered\"           \n",
      "[177] \"Un_gl000216 Chromosome not registered\"           \n",
      "[178] \"Un_gl000216 Chromosome not registered\"           \n",
      "[179] \"Un_gl000216 Chromosome not registered\"           \n",
      "[180] \"Un_gl000216 Chromosome not registered\"           \n",
      "[181] \"Un_gl000216 Chromosome not registered\"           \n",
      "[182] \"Un_gl000216 Chromosome not registered\"           \n",
      "[183] \"Un_gl000216 Chromosome not registered\"           \n",
      "[184] \"Un_gl000216 Chromosome not registered\"           \n",
      "[185] \"Un_gl000216 Chromosome not registered\"           \n",
      "[186] \"Un_gl000216 Chromosome not registered\"           \n",
      "[187] \"Un_gl000216 Chromosome not registered\"           \n",
      "[188] \"Un_gl000216 Chromosome not registered\"           \n",
      "[189] \"Un_gl000216 Chromosome not registered\"           \n",
      "[190] \"Un_gl000216 Chromosome not registered\"           \n",
      "[191] \"Un_gl000216 Chromosome not registered\"           \n",
      "[192] \"Un_gl000216 Chromosome not registered\"           \n",
      "[193] \"Un_gl000216 Chromosome not registered\"           \n",
      "[194] \"Un_gl000216 Chromosome not registered\"           \n",
      "[195] \"Un_gl000216 Chromosome not registered\"           \n",
      "[196] \"Un_gl000216 Chromosome not registered\"           \n",
      "[197] \"Un_gl000216 Chromosome not registered\"           \n",
      "[198] \"Un_gl000216 Chromosome not registered\"           \n",
      "[199] \"Un_gl000216 Chromosome not registered\"           \n",
      "[200] \"Un_gl000216 Chromosome not registered\"           \n",
      "[201] \"Un_gl000216 Chromosome not registered\"           \n",
      "[202] \"Un_gl000216 Chromosome not registered\"           \n",
      "[203] \"Un_gl000216 Chromosome not registered\"           \n",
      "[204] \"Un_gl000218 Chromosome not registered\"           \n",
      "[205] \"Un_gl000218 Chromosome not registered\"           \n",
      "[206] \"Un_gl000218 Chromosome not registered\"           \n",
      "[207] \"Un_gl000218 Chromosome not registered\"           \n",
      "[208] \"Un_gl000219 Chromosome not registered\"           \n",
      "[209] \"Un_gl000219 Chromosome not registered\"           \n",
      "[210] \"Un_gl000219 Chromosome not registered\"           \n",
      "[211] \"Un_gl000219 Chromosome not registered\"           \n",
      "[212] \"Un_gl000219 Chromosome not registered\"           \n",
      "[213] \"Un_gl000219 Chromosome not registered\"           \n",
      "[214] \"Un_gl000219 Chromosome not registered\"           \n",
      "[215] \"Un_gl000219 Chromosome not registered\"           \n",
      "[216] \"Un_gl000219 Chromosome not registered\"           \n",
      "[217] \"Un_gl000219 Chromosome not registered\"           \n",
      "[218] \"Un_gl000219 Chromosome not registered\"           \n",
      "[219] \"Un_gl000219 Chromosome not registered\"           \n",
      "[220] \"Un_gl000219 Chromosome not registered\"           \n",
      "[221] \"Un_gl000219 Chromosome not registered\"           \n",
      "[222] \"Un_gl000219 Chromosome not registered\"           \n",
      "[223] \"Un_gl000219 Chromosome not registered\"           \n",
      "[224] \"Un_gl000219 Chromosome not registered\"           \n",
      "[225] \"Un_gl000219 Chromosome not registered\"           \n",
      "[226] \"Un_gl000220 Chromosome not registered\"           \n",
      "[227] \"Un_gl000220 Chromosome not registered\"           \n",
      "[228] \"Un_gl000222 Chromosome not registered\"           \n",
      "[229] \"Un_gl000222 Chromosome not registered\"           \n",
      "[230] \"Un_gl000222 Chromosome not registered\"           \n",
      "[231] \"Un_gl000224 Chromosome not registered\"           \n",
      "[232] \"Un_gl000224 Chromosome not registered\"           \n",
      "[233] \"Un_gl000224 Chromosome not registered\"           \n",
      "[234] \"Un_gl000224 Chromosome not registered\"           \n",
      "[235] \"Un_gl000224 Chromosome not registered\"           \n",
      "[236] \"Un_gl000224 Chromosome not registered\"           \n",
      "[237] \"Un_gl000224 Chromosome not registered\"           \n",
      "[238] \"Un_gl000224 Chromosome not registered\"           \n",
      "[239] \"Un_gl000224 Chromosome not registered\"           \n",
      "[240] \"Un_gl000224 Chromosome not registered\"           \n",
      "[241] \"Un_gl000224 Chromosome not registered\"           \n",
      "[242] \"Un_gl000224 Chromosome not registered\"           \n",
      "[243] \"Un_gl000224 Chromosome not registered\"           \n",
      "[244] \"Un_gl000224 Chromosome not registered\"           \n",
      "[245] \"Un_gl000224 Chromosome not registered\"           \n",
      "[246] \"Un_gl000224 Chromosome not registered\"           \n",
      "[247] \"Un_gl000224 Chromosome not registered\"           \n",
      "[248] \"Un_gl000224 Chromosome not registered\"           \n",
      "[249] \"Un_gl000224 Chromosome not registered\"           \n",
      "[250] \"Un_gl000224 Chromosome not registered\"           \n",
      "[251] \"Un_gl000224 Chromosome not registered\"           \n",
      "[252] \"Un_gl000224 Chromosome not registered\"           \n",
      "[253] \"Un_gl000224 Chromosome not registered\"           \n",
      "[254] \"Un_gl000224 Chromosome not registered\"           \n",
      "[255] \"Un_gl000224 Chromosome not registered\"           \n",
      "[256] \"Un_gl000224 Chromosome not registered\"           \n",
      "[257] \"Un_gl000224 Chromosome not registered\"           \n",
      "[258] \"Un_gl000224 Chromosome not registered\"           \n",
      "[259] \"Un_gl000224 Chromosome not registered\"           \n",
      "[260] \"Un_gl000224 Chromosome not registered\"           \n",
      "[261] \"Un_gl000225 Chromosome not registered\"           \n",
      "[262] \"Un_gl000225 Chromosome not registered\"           \n",
      "[263] \"Un_gl000225 Chromosome not registered\"           \n",
      "[264] \"Un_gl000225 Chromosome not registered\"           \n",
      "[265] \"Un_gl000225 Chromosome not registered\"           \n",
      "[266] \"Un_gl000225 Chromosome not registered\"           \n",
      "[267] \"Un_gl000225 Chromosome not registered\"           \n",
      "[268] \"Un_gl000225 Chromosome not registered\"           \n",
      "[269] \"Un_gl000225 Chromosome not registered\"           \n",
      "[270] \"Un_gl000225 Chromosome not registered\"           \n",
      "[271] \"Un_gl000225 Chromosome not registered\"           \n",
      "[272] \"Un_gl000225 Chromosome not registered\"           \n",
      "[273] \"Un_gl000225 Chromosome not registered\"           \n",
      "[274] \"Un_gl000225 Chromosome not registered\"           \n",
      "[275] \"Un_gl000225 Chromosome not registered\"           \n",
      "[276] \"Un_gl000225 Chromosome not registered\"           \n",
      "[277] \"Un_gl000225 Chromosome not registered\"           \n",
      "[278] \"Un_gl000225 Chromosome not registered\"           \n",
      "[279] \"Un_gl000225 Chromosome not registered\"           \n",
      "[280] \"Un_gl000225 Chromosome not registered\"           \n",
      "[281] \"Un_gl000225 Chromosome not registered\"           \n",
      "[282] \"Un_gl000225 Chromosome not registered\"           \n",
      "[283] \"Un_gl000225 Chromosome not registered\"           \n",
      "[284] \"Un_gl000225 Chromosome not registered\"           \n",
      "[285] \"Un_gl000225 Chromosome not registered\"           \n",
      "[286] \"Un_gl000225 Chromosome not registered\"           \n",
      "[287] \"Un_gl000225 Chromosome not registered\"           \n",
      "[288] \"Un_gl000225 Chromosome not registered\"           \n",
      "[289] \"Un_gl000225 Chromosome not registered\"           \n",
      "[290] \"Un_gl000225 Chromosome not registered\"           \n",
      "[291] \"Un_gl000225 Chromosome not registered\"           \n",
      "[292] \"Un_gl000225 Chromosome not registered\"           \n",
      "[293] \"Un_gl000225 Chromosome not registered\"           \n",
      "[294] \"Un_gl000225 Chromosome not registered\"           \n",
      "[295] \"Un_gl000225 Chromosome not registered\"           \n",
      "[296] \"Un_gl000225 Chromosome not registered\"           \n",
      "[297] \"Un_gl000226 Chromosome not registered\"           \n",
      "[298] \"Un_gl000226 Chromosome not registered\"           \n",
      "[299] \"Un_gl000226 Chromosome not registered\"           \n",
      "[300] \"Un_gl000228 Chromosome not registered\"           \n",
      "[301] \"Un_gl000228 Chromosome not registered\"           \n",
      "[302] \"Un_gl000229 Chromosome not registered\"           \n",
      "[303] \"Un_gl000229 Chromosome not registered\"           \n",
      "[304] \"Un_gl000229 Chromosome not registered\"           \n",
      "[305] \"Un_gl000229 Chromosome not registered\"           \n",
      "[306] \"Un_gl000229 Chromosome not registered\"           \n",
      "[307] \"Un_gl000230 Chromosome not registered\"           \n",
      "[308] \"Un_gl000230 Chromosome not registered\"           \n",
      "[309] \"Un_gl000230 Chromosome not registered\"           \n",
      "[310] \"Un_gl000230 Chromosome not registered\"           \n",
      "[311] \"Un_gl000230 Chromosome not registered\"           \n",
      "[312] \"Un_gl000230 Chromosome not registered\"           \n",
      "[313] \"Un_gl000230 Chromosome not registered\"           \n",
      "[314] \"Un_gl000230 Chromosome not registered\"           \n",
      "[315] \"Un_gl000230 Chromosome not registered\"           \n",
      "[316] \"Un_gl000230 Chromosome not registered\"           \n",
      "[317] \"Un_gl000230 Chromosome not registered\"           \n",
      "[318] \"Un_gl000230 Chromosome not registered\"           \n",
      "[319] \"Un_gl000231 Chromosome not registered\"           \n",
      "[320] \"Un_gl000231 Chromosome not registered\"           \n",
      "[321] \"Un_gl000231 Chromosome not registered\"           \n",
      "[322] \"Un_gl000231 Chromosome not registered\"           \n",
      "[323] \"Un_gl000231 Chromosome not registered\"           \n",
      "[324] \"Un_gl000231 Chromosome not registered\"           \n",
      "[325] \"Un_gl000231 Chromosome not registered\"           \n",
      "[326] \"Un_gl000232 Chromosome not registered\"           \n",
      "[327] \"Un_gl000232 Chromosome not registered\"           \n",
      "[328] \"Un_gl000232 Chromosome not registered\"           \n",
      "[329] \"Un_gl000232 Chromosome not registered\"           \n",
      "[330] \"Un_gl000232 Chromosome not registered\"           \n",
      "[331] \"Un_gl000233 Chromosome not registered\"           \n",
      "[332] \"Un_gl000233 Chromosome not registered\"           \n",
      "[333] \"Un_gl000233 Chromosome not registered\"           \n",
      "[334] \"Un_gl000233 Chromosome not registered\"           \n",
      "[335] \"Un_gl000233 Chromosome not registered\"           \n",
      "[336] \"Un_gl000233 Chromosome not registered\"           \n",
      "[337] \"Un_gl000234 Chromosome not registered\"           \n",
      "[338] \"Un_gl000234 Chromosome not registered\"           \n",
      "[339] \"Un_gl000234 Chromosome not registered\"           \n",
      "[340] \"Un_gl000234 Chromosome not registered\"           \n",
      "[341] \"Un_gl000234 Chromosome not registered\"           \n",
      "[342] \"Un_gl000234 Chromosome not registered\"           \n",
      "[343] \"Un_gl000234 Chromosome not registered\"           \n",
      "[344] \"Un_gl000234 Chromosome not registered\"           \n",
      "[345] \"Un_gl000234 Chromosome not registered\"           \n",
      "[346] \"Un_gl000234 Chromosome not registered\"           \n",
      "[347] \"Un_gl000235 Chromosome not registered\"           \n",
      "[348] \"Un_gl000235 Chromosome not registered\"           \n",
      "[349] \"Un_gl000235 Chromosome not registered\"           \n",
      "[350] \"Un_gl000235 Chromosome not registered\"           \n",
      "[351] \"Un_gl000235 Chromosome not registered\"           \n",
      "[352] \"Un_gl000235 Chromosome not registered\"           \n",
      "[353] \"Un_gl000241 Chromosome not registered\"           \n",
      "[354] \"Un_gl000241 Chromosome not registered\"           \n",
      "[355] \"Un_gl000241 Chromosome not registered\"           \n",
      "[356] \"Un_gl000241 Chromosome not registered\"           \n",
      "[357] \"Un_gl000241 Chromosome not registered\"           \n",
      "[358] \"Un_gl000241 Chromosome not registered\"           \n",
      "[359] \"Un_gl000241 Chromosome not registered\"           \n",
      "[360] \"Finish Annotation\"                               \n",
      "[361] \"python process end: 2020-01-06 10:30:01.604672\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2101439.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " [1] \"python process start: 2020-01-06 10:33:07.770147\"\n",
      " [2] \"Load Reference\"                                  \n",
      " [3] \"Check Reference files\"                           \n",
      " [4] \"fixed reference done: 2020-01-06 10:33:28.521632\"\n",
      " [5] \"Start Annotation\"                                \n",
      " [6] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      " [7] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      " [8] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      " [9] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[10] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[11] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[12] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[13] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[14] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[15] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[16] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[17] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[18] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[19] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[20] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[21] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[22] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[23] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[24] \"17_ctg5_hap1 Chromosome not registered\"          \n",
      "[25] \"17_gl000204_random Chromosome not registered\"    \n",
      "[26] \"17_gl000205_random Chromosome not registered\"    \n",
      "[27] \"17_gl000205_random Chromosome not registered\"    \n",
      "[28] \"17_gl000205_random Chromosome not registered\"    \n",
      "[29] \"17_gl000205_random Chromosome not registered\"    \n",
      "[30] \"17_gl000205_random Chromosome not registered\"    \n",
      "[31] \"17_gl000205_random Chromosome not registered\"    \n",
      "[32] \"17_gl000205_random Chromosome not registered\"    \n",
      "[33] \"17_gl000205_random Chromosome not registered\"    \n",
      "[34] \"17_gl000205_random Chromosome not registered\"    \n",
      "[35] \"19_gl000209_random Chromosome not registered\"    \n",
      "[36] \"19_gl000209_random Chromosome not registered\"    \n",
      "[37] \"1_gl000192_random Chromosome not registered\"     \n",
      "[38] \"1_gl000192_random Chromosome not registered\"     \n",
      "[39] \"4_gl000193_random Chromosome not registered\"     \n",
      "[40] \"4_gl000193_random Chromosome not registered\"     \n",
      "[41] \"4_gl000194_random Chromosome not registered\"     \n",
      "[42] \"4_gl000194_random Chromosome not registered\"     \n",
      "[43] \"4_gl000194_random Chromosome not registered\"     \n",
      "[44] \"4_gl000194_random Chromosome not registered\"     \n",
      "[45] \"4_gl000194_random Chromosome not registered\"     \n",
      "[46] \"4_gl000194_random Chromosome not registered\"     \n",
      "[47] \"4_gl000194_random Chromosome not registered\"     \n",
      "[48] \"4_gl000194_random Chromosome not registered\"     \n",
      "[49] \"4_gl000194_random Chromosome not registered\"     \n",
      "[50] \"4_gl000194_random Chromosome not registered\"     \n",
      "[51] \"4_gl000194_random Chromosome not registered\"     \n",
      "[52] \"4_gl000194_random Chromosome not registered\"     \n",
      "[53] \"4_gl000194_random Chromosome not registered\"     \n",
      "[54] \"4_gl000194_random Chromosome not registered\"     \n",
      "[55] \"6_mcf_hap5 Chromosome not registered\"            \n",
      "[56] \"6_ssto_hap7 Chromosome not registered\"           \n",
      "[57] \"7_gl000195_random Chromosome not registered\"     \n",
      "[58] \"7_gl000195_random Chromosome not registered\"     \n",
      "[59] \"7_gl000195_random Chromosome not registered\"     \n",
      "[60] \"7_gl000195_random Chromosome not registered\"     \n",
      "[61] \"7_gl000195_random Chromosome not registered\"     \n",
      "[62] \"Un_gl000211 Chromosome not registered\"           \n",
      "[63] \"Un_gl000211 Chromosome not registered\"           \n",
      "[64] \"Un_gl000211 Chromosome not registered\"           \n",
      "[65] \"Un_gl000214 Chromosome not registered\"           \n",
      "[66] \"Un_gl000214 Chromosome not registered\"           \n",
      "[67] \"Un_gl000214 Chromosome not registered\"           \n",
      "[68] \"Un_gl000214 Chromosome not registered\"           \n",
      "[69] \"Un_gl000218 Chromosome not registered\"           \n",
      "[70] \"Un_gl000218 Chromosome not registered\"           \n",
      "[71] \"Un_gl000218 Chromosome not registered\"           \n",
      "[72] \"Un_gl000218 Chromosome not registered\"           \n",
      "[73] \"Un_gl000218 Chromosome not registered\"           \n",
      "[74] \"Un_gl000218 Chromosome not registered\"           \n",
      "[75] \"Un_gl000218 Chromosome not registered\"           \n",
      "[76] \"Un_gl000218 Chromosome not registered\"           \n",
      "[77] \"Un_gl000219 Chromosome not registered\"           \n",
      "[78] \"Un_gl000219 Chromosome not registered\"           \n",
      "[79] \"Un_gl000219 Chromosome not registered\"           \n",
      "[80] \"Un_gl000219 Chromosome not registered\"           \n",
      "[81] \"Un_gl000219 Chromosome not registered\"           \n",
      "[82] \"Un_gl000219 Chromosome not registered\"           \n",
      "[83] \"Un_gl000224 Chromosome not registered\"           \n",
      "[84] \"Un_gl000224 Chromosome not registered\"           \n",
      "[85] \"Un_gl000229 Chromosome not registered\"           \n",
      "[86] \"Un_gl000231 Chromosome not registered\"           \n",
      "[87] \"Un_gl000232 Chromosome not registered\"           \n",
      "[88] \"Finish Annotation\"                               \n",
      "[89] \"python process end: 2020-01-06 10:33:39.472372\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSM2298950.bed\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"python process start: 2020-01-06 10:38:38.928283\"\n",
      "[2] \"Load Reference\"                                  \n",
      "[3] \"Check Reference files\"                           \n",
      "[4] \"fixed reference done: 2020-01-06 10:38:58.284831\"\n",
      "[5] \"Start Annotation\"                                \n",
      "[6] \"Finish Annotation\"                               \n",
      "[7] \"python process end: 2020-01-06 10:38:58.445859\"  \n",
      "[1] \"Start test..............\"\n",
      "[1] \"Fisher's exact test done\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading peaks from ./testData/GSE84618.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM1847178.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2058021.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000028 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000042 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000289 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000460 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000466 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000469 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000715 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000722 with 36 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000729 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0001522 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0001844 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0001977 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002183 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002190 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002313 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002467 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006085 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006086 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006266 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006271 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006297 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006488 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006490 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006506 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006607 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006654 with 37 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006658 with 30 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006895 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006999 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0007064 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0007094 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0007095 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0007220 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009081 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009083 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009219 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0010869 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0010939 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015693 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016075 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016180 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016242 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016254 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016578 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019076 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019372 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0021904 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030033 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030277 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031033 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031293 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031440 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031468 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031577 with 36 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032042 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032288 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032469 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032516 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032528 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032786 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032878 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033119 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033260 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033522 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033598 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033599 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033866 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033962 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034030 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034033 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034123 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034204 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034453 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034501 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035269 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035635 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035886 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035890 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035891 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036148 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036149 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036150 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036151 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036152 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036474 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036475 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0038202 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042149 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042276 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042448 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042559 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042762 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042769 with 36 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043162 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043555 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043558 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043649 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043968 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044090 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044786 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045324 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045332 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045603 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045879 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046112 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046471 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046473 with 37 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046825 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046827 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046856 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048025 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048308 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048488 with 30 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050686 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050687 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051204 with 33 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051383 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051782 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051895 with 15 genes...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Applying correction for geneset GO:0051973 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052126 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052192 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060049 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060390 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060544 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0061157 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070200 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070242 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070584 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070828 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070987 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071173 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071174 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071294 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071459 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071711 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072600 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0080182 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090136 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090140 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090141 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090231 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090503 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097035 with 37 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097150 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097320 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097734 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900739 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900740 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900745 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:1901030 with 37 genes...\n",
      "\n",
      "Applying correction for geneset GO:1901032 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902001 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902186 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902187 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902188 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903203 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903204 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903205 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903206 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903312 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903392 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903432 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903541 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903543 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903649 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903830 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904353 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904355 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904357 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:1990182 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000104 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000114 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000641 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:2001267 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:2001269 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:2001273 with 19 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000407 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005662 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005852 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0018995 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030014 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030894 with 30 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035145 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035267 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035327 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035371 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0038201 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042575 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043034 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043189 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043596 with 43 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043601 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043657 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071782 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072546 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090723 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1990752 with 22 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000175 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0001103 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0003841 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0003887 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004012 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004033 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004532 with 33 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004659 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004675 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004693 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005001 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005123 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005521 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005548 with 49 genes...\n",
      "\n",
      "Applying correction for geneset GO:0008106 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0008191 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015095 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015923 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016628 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016722 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016799 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016866 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016879 with 49 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016881 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016896 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016922 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0017049 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019198 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030515 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031435 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034061 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034237 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034593 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042171 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046965 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048027 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048531 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050431 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050664 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051019 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051537 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051539 with 40 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052745 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052866 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070064 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070182 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070273 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070300 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071617 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097472 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098641 with 18 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset hsa00531 with 18 genes...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Applying correction for geneset hsa00532 with 22 genes...\n",
      "\n",
      "Applying correction for geneset hsa03410 with 33 genes...\n",
      "\n",
      "Applying correction for geneset hsa03430 with 23 genes...\n",
      "\n",
      "Applying correction for geneset hsa04710 with 22 genes...\n",
      "\n",
      "Applying correction for geneset hsa05213 with 52 genes...\n",
      "\n",
      "Reading peaks from ./testData/GSM2058022.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000028 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000042 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000289 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002183 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002190 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006098 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006144 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006271 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006488 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006490 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006506 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006999 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009081 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009083 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009219 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0010390 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0010804 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016254 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019682 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0021783 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030277 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031293 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031468 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031958 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032506 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032516 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032786 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033119 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033522 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033866 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033962 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034030 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034033 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034453 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034501 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042921 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043162 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044090 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045022 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045603 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046599 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048025 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048308 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051383 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051782 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051895 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070584 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070828 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070932 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070987 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071294 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072600 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090231 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097734 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098901 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098927 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902188 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903392 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903541 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903543 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1990182 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000104 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000641 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:2001269 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:2001273 with 19 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0005662 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005852 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0018995 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030014 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0038201 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043657 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044215 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044216 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044217 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071782 with 16 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0004659 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004675 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005521 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015923 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0017136 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031435 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032183 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034483 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034979 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046965 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046966 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050431 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051019 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052745 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070064 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070182 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070273 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070300 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098641 with 18 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset hsa03430 with 23 genes...\n",
      "\n",
      "Applying correction for geneset hsa04710 with 22 genes...\n",
      "\n",
      "Reading peaks from ./testData/GSM2058023.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000027 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000028 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000460 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000466 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000469 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002183 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002190 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002313 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002467 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006085 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006086 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006098 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006271 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006506 with 31 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006907 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006999 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0008053 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0008356 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009303 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0009312 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0010939 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0014912 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016254 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016578 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0018126 with 20 genes...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Applying correction for geneset GO:0019372 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019682 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0021756 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030277 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031112 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031116 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031293 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031440 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032469 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032516 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032786 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032878 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032922 with 55 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033119 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033522 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033866 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034030 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034033 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034123 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034453 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034501 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035635 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035886 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0036003 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0038066 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042026 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042149 with 34 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042559 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042762 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042776 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043162 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0044090 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045022 with 35 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045603 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046475 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046827 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046885 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048025 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048308 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050686 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051383 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051457 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051782 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051898 with 36 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060049 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060444 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060544 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070828 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071243 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071294 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071459 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071711 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072595 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072673 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0080182 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090136 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090185 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090231 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098901 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098927 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900101 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900744 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:1900745 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902001 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1902187 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903312 with 38 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904357 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000114 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000641 with 17 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000407 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005662 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005753 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005852 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0018995 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030867 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035145 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035371 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0038201 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043657 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0045259 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070822 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071782 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0090723 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1990752 with 22 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000400 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0003756 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004659 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004675 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005123 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005521 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015923 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016706 with 46 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016864 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016879 with 49 genes...\n",
      "\n",
      "Applying correction for geneset GO:0016881 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0017049 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0017134 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0019789 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031435 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033612 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034237 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034483 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046933 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046965 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048531 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050431 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0050998 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051019 with 23 genes...\n",
      "\n",
      "Applying correction for geneset GO:0051537 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0052745 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0055106 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070064 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070273 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070300 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0097718 with 30 genes...\n",
      "\n",
      "Applying correction for geneset GO:0098641 with 18 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset hsa00450 with 17 genes...\n",
      "\n",
      "Applying correction for geneset hsa00900 with 15 genes...\n",
      "\n",
      "Reading peaks from ./testData/GSM2101436.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101437.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000028 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0000470 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002183 with 22 genes...\n",
      "\n",
      "Applying correction for geneset GO:0002190 with 15 genes...\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Applying correction for geneset GO:0002756 with 33 genes...\n",
      "\n",
      "Applying correction for geneset GO:0003222 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006388 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006743 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0006744 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0031958 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032011 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032012 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0032469 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0033866 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034030 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034033 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034315 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0034453 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035561 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035635 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035666 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035735 with 40 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042451 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0042921 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0043536 with 27 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046129 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048308 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0048569 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0055070 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0060055 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0061157 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070200 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070584 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0071498 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0072661 with 24 genes...\n",
      "\n",
      "Applying correction for geneset GO:1901663 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:1903077 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904353 with 26 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904355 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:1904376 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:2000811 with 16 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0005680 with 20 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005685 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0005697 with 21 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030990 with 28 genes...\n",
      "\n",
      "Applying correction for geneset GO:0030992 with 18 genes...\n",
      "\n",
      "Applying correction for geneset GO:0035371 with 17 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070822 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:1990752 with 22 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset GO:0000400 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0004659 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0015923 with 15 genes...\n",
      "\n",
      "Applying correction for geneset GO:0046875 with 25 genes...\n",
      "\n",
      "Applying correction for geneset GO:0061631 with 29 genes...\n",
      "\n",
      "Applying correction for geneset GO:0061650 with 30 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070034 with 16 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070064 with 19 genes...\n",
      "\n",
      "Applying correction for geneset GO:0070182 with 15 genes...\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Applying correction for geneset hsa04710 with 22 genes...\n",
      "\n",
      "Reading peaks from ./testData/GSM2101438.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101439.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2298950.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peaks(...) ..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: ChIP-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSE84618.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM1847178.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2058021.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2058022.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2058023.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101436.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101437.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101438.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2101439.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Reading peaks from ./testData/GSM2298950.bed\n",
      "\n",
      "Assigning peaks to genes with assign_peak_segments(...) ..\n",
      "\n",
      "Calculating peak overlaps with gene loci..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Biological Process\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Cellular Component\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: Gene Ontology Molecular Function\n",
      "\n",
      "Running tests..\n",
      "\n",
      "Test: Broad-Enrich\n",
      "\n",
      "Genesets: KEGG Pathways\n",
      "\n",
      "Running tests..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"./protocolFunctions/executeChipenrichBroadenrichSeq2pathway.R\")\n",
    "executeChipenrichBroadenrichSeq2pathway(\"./testData/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pruning Results from Chipenrich, Broadenrich, and Seq2pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since we shall majorly be interested in the enrichment terms and their statistical significance values, we'll extract these from the other attributes of the results from the tools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "source(\"./protocolFunctions/extractingValuedResults.R\")\n",
    "extractingValuedResults()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now that we have the results from Chipenrich, Broadenrich, and Seq2pathway, we move towards framing results from Enrichr and GREAT that we have manually compiled. Just to reiterate, both the tools are available as a web-interface as of now. (GREAT: http://bejerano.stanford.edu/great/public/html/index.php, Enrichr: http://amp.pharm.mssm.edu/Enrichr/)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Structuring Results from Enrichr and GREAT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Enrichr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "source(\"./protocolFunctions/enrichrResultsCompilation.R\")\n",
    "enrichrResultsCompilation()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "At this point, we have compiled results from different databases of Enrichr, viz. GO : Cellular Component, Molecular Function, Biological Process, and KEGG. For computational convenience, we shall combine all these results together for each sample in our benchmark dataset. Another task to perform here is to extract GO and KEGG ids from the enrichment terms. The results contain merged information and we prune the term definition."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub data rate exceeded.\n",
      "The notebook server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--NotebookApp.iopub_data_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)\n",
      "NotebookApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "source(\"./protocolFunctions/enrichrPostprocessing.R\")\n",
    "enrichrPostprocessing()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### GREAT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
